spring-cloud-task-core/src/main/java/org/springframework/cloud/task/configuration/SimpleTaskConfiguration.java

```diff
@@ -34,12 +39,22 @@
  *
  * @author Glenn Renfro
  */
+@EnableTransactionManagement
 @Configuration
 public class SimpleTaskConfiguration {
 
 	@Autowired
 	private ApplicationContext context;
 
+	@Autowired(required = false)
+	private Collection<DataSource> dataSources;
```

Comment:
To your first point, if the user wants a setter, then they should probably implement their own configurer.  
To the second point, I'll raise a github issue and let's discuss.